clippy_lints/src/assign_ops.rs

```diff
@@ -135,6 +135,19 @@ impl LateLintPass for AssignOps {
                                         } else {
                                             return; // useless if the trait doesn't exist
                                         };
+                                        // check that we are not inside an `impl AssignOp` of this exact operation
+                                        let parent_fn = cx.tcx.map.get_parent(e.id);
+                                        let parent_impl = cx.tcx.map.get_parent(parent_fn);
+                                        if parent_impl.as_u32() != 0 {
```

Comment:
without this check, the `get` call in the next line will panic. All other functions on `map` will also panic, so there's no `Option` way to do this correctly. Should probably fix this in rustc...